packages/core/core-flows/src/order/workflows/return/confirm-receive-return-request.ts

```diff
@@ -40,13 +48,95 @@ const validationStep = createStep(
   }
 )
 
+function prepareInventoryUpdate({ orderReturn, returnedQuantityMap }) {
+  const inventoryAdjustment: {
+    inventory_item_id: string
+    location_id: string
+    adjustment: BigNumberInput
+  }[] = []
+
+  let hasManagedInventory = false
+  let hasStockLocation = false
+
+  const productVariantInventoryItems = new Map<string, any>()
+
+  deepFlatMap(
+    orderReturn.items,
+    "item.variant.inventory_items.inventory.location_levels",
+    ({ variant, inventory_items, location_levels }) => {
+      if (!variant?.manage_inventory) {
+        return
+      }
+      hasManagedInventory = true
+
+      if (location_levels?.location_id !== orderReturn.location_id) {
+        return
+      }
+      hasStockLocation = true
+
+      if (!inventory_items) {
+        return
+      }
+
+      const inventoryItemId = inventory_items.inventory_item_id
+
+      if (!productVariantInventoryItems.has(inventoryItemId)) {
+        productVariantInventoryItems.set(inventoryItemId, {
+          variant_id: inventory_items.variant_id,
+          inventory_item_id: inventoryItemId,
+          required_quantity: inventory_items.required_quantity,
+        })
+      }
+    }
+  )
+
+  if (hasManagedInventory && !hasStockLocation) {
+    // TODO: how to bypass this error?
```

Comment:
question: This looks OK at first glance. What have you considered here?

Reply:
Just thinking if someone get stuck because of this condition, if they could somehow bypass it. But maybe they need to update the location of the return and then confirm this.